lib/streamlit/runtime/script_data.py

```diff
@@ -24,7 +23,7 @@
 
 
 @dataclass(init=False)
-class SessionData:
+class ScriptData:
```

Comment:
This is a great change! Is it possible to make it a `frozen=True` dataclass now, or do we mutate it?

Reply:
It's not mutated as far as I know / doing a quick search over the codebase -- will also make this frozen ❄️

Reply:
Argh, maybe this isn't so easy as apparently setting `frozen=True` also prevents you from setting the value of a dataclass field from within its `__init__` method.

A workaround would be to add a `create` or `new` staticmethod that does what we currently do in `__init__`, then use that method to create new `ScriptData` objects, but I'm hesitant to do that because that pattern isn't common in the Python world as far as I know

Reply:
I'm not opposed to that, FWIW (but I don't feel super strongly either).

Reply:
Found a reasonable/more Pythonic way of doing this 🎉 

Apparently you can use `object.__setattr__` to be able to set fields anyway in a frozen dataclass (the way that frozen dataclasses work under the hood is apparently to overwrite `self.__setattr__` to throw an exception, so using `object.__set_attr__` bypasses this), and the `__post_init__` magic method can be used to use the auto-generated dataclass `__init__` + do things afterwards